Patch supermini

dtashmore547 opened the pull request:
Update to platformio.ini to allow the ESP32C3 SuperMini board to work along with the onboard blue LED.

The following Files were added to the .platformio folder:-

1)  in folder ~/.platformio/packages/framework-arduinoespressif32/variants/super_mini_esp32c3

[pins.arduino_super_mni_esp32c3.zip](https://github.com/user-attachments/files/18363959/pins.arduino_super_mini_esp32c3.zip) 

2) in folder ~/.platformio/platforms/espressif32/boards

[super_mini_esp32c3.json](https://github.com/user-attachments/files/18364032/super_mini_esp32c3.json)

On the original Ispindel board the super mini board is small enough to be stuck onto the board by double sided 3M Pads and fly wired to the correct connections.
I have a PCB available on PCBWay.com mainly intended for the Rapt Pill housing but should also fit the ISpindel tube.

mp-se commented:
Hi, thanks for the contribution but in order to merge this to the codebase please correct the following:

* I'm using automated builds via github actions and there is no way to manually add the defintions to make a build. Please create the needed scripts that add the board definitions to arduino before the build so the automated build process will work.
* Since this is a modified version i want to have some updates to the documentation that describe this target as well as links to where detailed documentation and the pcb can be found so that those who wants to use this can make the build. I dont need to include the full details in gravitymon just a placeholder to where details can be found. 
* I'm fine with supporting multiple hardware versions (ispindel and floaty) and this could be seen as a separate hardware setup if you have your own pcb and build instructions. There is a return value in the /api/status that can be used to indicate what hardware gravitymon is running on. Whats your view on this ? is this a standard ispindle hardware or another variant?

Update your own branch and any changes there should automatically be reflected in this PR.

dtashmore547 commented:
hi Magnus
sorry for the delay, it may be a while before i can correct my error, i am going into hospital for a shoulder op, i will continue when i am able, but sure what i did wrong as i am inexperienced with GitHub but will endeavour to put it right, good learning experience! 
regards
Dave

On 10 January 2025 09:58:43 GMT+02:00, Magnus ***@***.***> wrote:
>Hi, thanks for the contribution but in order to merge this to the codebase please correct the following:
>
>* I'm using automated builds via github actions and there is no way to manually add the defintions to make a build. Please create the needed scripts that add the board definitions to arduino before the build so the automated build process will work.
>* Since this is a modified version i want to have some updates to the documentation that describe this target as well as links to where detailed documentation and the pcb can be found so that those who wants to use this can make the build. I dont need to include the full details in gravitymon just a placeholder to where details can be found. 
>* I'm fine with supporting multiple hardware versions (ispindel and floaty) and this could be seen as a separate hardware setup if you have your own pcb and build instructions. There is a return value in the /api/status that can be used to indicate what hardware gravitymon is running on. Whats your view on this ? is this a standard ispindle hardware or another variant?
>
>Update your own branch and any changes there should automatically be reflected in this PR.
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/mp-se/gravitymon/pull/205#issuecomment-2581997541
>You are receiving this because you authored the thread.
>
>Message ID: ***@***.***>
-- 
Sent from my Android phone with K-9 Mail. Please excuse my brevity.

mp-se commented:
No worries, its just a few more things that needs to be added for it to work properly. Let me know if you need any help with the tasks. You have some example in the thread for how to add the arduino board. Let me know if you need more info

dtashmore547 commented:
will do, should have time whilst recovering and little else to do.

On 15 January 2025 09:17:23 GMT+02:00, Magnus ***@***.***> wrote:
>No worries, its just a few more things that needs to be added for it to work properly. Let me know if you need any help with the tasks. You have some example in the thread for how to add the arduino board. Let me know if you need more info
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/mp-se/gravitymon/pull/205#issuecomment-2591810194
>You are receiving this because you authored the thread.
>
>Message ID: ***@***.***>
-- 
Sent from my Android phone with K-9 Mail. Please excuse my brevity.

mp-se commented:
Hi! Since i'm getting more requests for custom boards I have done some refactor to better support this aspect. 

* The board ID will now be used for the PIN configuration in the main.hpp file. A script will inject a define with the board ID so that can be used to map the pins correctly. See: scripts/board.py
* Firmware script will now also copy the file and name it custom-board.bin unless its one of the targets that I can test. This way its visible which ones are custom. So adding a custom board should not require any changes to that script.
*One thing remains to be adressed and that is the OTA support since that is in the espframework, I need to change that so the filename can be set from gravitymon. I will also look into how to add this to the documentation.

In you case we also need to patch the arduino framework with your new board and you can use the patch_arduino.py script as a base for that. You can put the board files in a directory called patch/<board-id>

Please have a look at the changes and give me some feedback.

mp-se commented:
I've added a script to patch platformio with new board defintions to the dev branch. Just copy the files to the correct location in the patch directory, script is run before a build

dtashmore547 commented:
hi Magnus
Great sounds like you have been very busy, i am a one arm bandit at present, had a shoulder op but will check out the new build as soon as i can
best regards
Dave

On 17 January 2025 19:02:27 GMT+02:00, Magnus ***@***.***> wrote:
>I've added a script to patch platformio with new board defintions to the dev branch. Just copy the files to the correct location in the patch directory, script is run before a build
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/mp-se/gravitymon/pull/205#issuecomment-2598797142
>You are receiving this because you authored the thread.
>
>Message ID: ***@***.***>
-- 
Sent from my Android phone with K-9 Mail. Please excuse my brevity.

mp-se commented:
> hi Magnus Great sounds like you have been very busy, i am a one arm bandit at present, had a shoulder op but will check out the new build as soon as i can best regards Dave
> On 17 January 2025 19:02:27 GMT+02:00, Magnus ***@***.***> wrote: I've added a script to patch platformio with new board defintions to the dev branch. Just copy the files to the correct location in the patch directory, script is run before a build -- Reply to this email directly or view it on GitHub: [#205 (comment)](https://github.com/mp-se/gravitymon/pull/205#issuecomment-2598797142) You are receiving this because you authored the thread. Message ID: ***@***.***>
> -- Sent from my Android phone with K-9 Mail. Please excuse my brevity.

It was not that much work, since i already had some code lying around it would be a quick task to add that possibility. I tested with your files so if you put them in the right folder and adjust the main.hpp file to match the new structure it should be a quick update for you. You need to do a new pull from my dev branch to integrate my changes before, when you do a new push the PR should be updated automatically with your changes.

I also added a file src_docs/custom.rst where you can add the documentation and links to where more information can be found. 

I've also added some instructions in the CONTRIBUTE.md file on what steps are needed to add a new custom board... 

Its not a rush, make sure your shoulder is healed and take care of yourself.

mp-se commented:
I merged your custom board and added the needed settings as per the new standard. Please test the version for your board and open a new PR that updates the docs for you board.

dtashmore547 commented:
hi Magnus
I have tested the new build but it won't build for me, I deleted my own modifications to the platform and downloaded the zip which I opened in platformio it built ok but when I changed the board to super_mini_esp32c3 it failed on [.pio/build/gravity32c3-release/src/ble.cpp.o] error 1
it looks like the board definition now conflicts with the ble settings.
when I made the initial changes I didn't have any conflicts during the build, I don't change any of the original pin configurations, I only make the 2 changes in the platform.ini file (board definition and re-enable the led, in all other aspects my  board is the same as the original Ispindel boards just changed to accommodate the different storm factor of the super mini, the circuitry is the same, so I didn't change any pin defines of the other c3 board you use, basically the super mini is the same as the d1 mini you already use with the exception of dio8 connected to the on-board led and only having a single blue led instead of a multi coloured led.
regards
Dave

On 20 January 2025 21:50:25 GMT+02:00, Magnus ***@***.***> wrote:
>I merged your custom board and added the needed settings as per the new standard. Please test the version for your board and open a new PR that updates the docs for you board. 
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/mp-se/gravitymon/pull/205#issuecomment-2603142720
>You are receiving this because you authored the thread.
>
>Message ID: ***@***.***>
-- 
Sent from my Android phone with K-9 Mail. Please excuse my brevity.

mp-se commented:
I had to disable ble for it to work on my end. I have updated the ble library so there might be some conflict there, you could try with a older version to see if it works, also check the board defintions since your link didnt work and i dowloaded it from your link in the discussion

mp-se commented:
Check that you have pulled the dev branch correctly because the automated build complies but i cannot test it since i dont have the board

dtashmore547 commented:
i will redo thanks, i will also check disabling ble though i would like this as i am awaiting parts to build the Bluetooth bridge. will have to wait until tomorrow as my shoulder is nagging at me now, doesn't take much.
regards
Dave

On 22 January 2025 13:07:32 GMT+02:00, Magnus ***@***.***> wrote:
>Check that you have pulled the dev branch correctly because the automated build complies but i cannot test it since i dont have the board
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/mp-se/gravitymon/pull/205#issuecomment-2606943765
>You are receiving this because you authored the thread.
>
>Message ID: ***@***.***>
-- 
Sent from my Android phone with K-9 Mail. Please excuse my brevity.

dtashmore547 commented:
good morning Magnus
I have the dev branch working on my esp32c3 super mini build, for some reason the clone ofthe dev branch was only lagging the master branch, I downloaded the zip file and it is working with the ble commented out in main.
do you have any clues as to why the bluetooth will not work? would be nice to be able to utilize the bluetooth bridge you have designed.
I have a minor issue with my custom board for three c3 super mini, initially I brought a charger receiver for my wireless design and ordered a few of the receiver chips for the board design, everything was working nicely but recently I brought an off the shelf charger and it doesn't work, it pulses for some reason, I have brought a few different charger boards with the same result, I cannot find the original charger board as it was too long ago, I have to try and find one that works to include in my design before I can use it as a complete design, a bit hampered by our badly broken postal service so it may take some time. there is no problem attaching the super mini board onto an ispindel board (with glue or double sided tape) and fly wiring it as the small size gives lots of room. alternatively my board does work fine without the wireless if it is used in an ispindel tube, my base intention has always to use the rapt pill housing as it works well under pressure but is difficult to open and close making wireless charging desirable. should I wait for getting the charger sorted or make available sooner. what is your opinion.
regards
Dave

On 22 January 2025 13:07:32 GMT+02:00, Magnus ***@***.***> wrote:
>Check that you have pulled the dev branch correctly because the automated build complies but i cannot test it since i dont have the board
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/mp-se/gravitymon/pull/205#issuecomment-2606943765
>You are receiving this because you authored the thread.
>
>Message ID: ***@***.***>
-- 
Sent from my Android phone with K-9 Mail. Please excuse my brevity.

dtashmore547 commented:
good day Magnus
I am finally back on the job after a few weeks and two shoulder operations. 
I am having some difficulty with pushing to grainfather, have you any experience with this site? I have brewspy and brewfather working on my c3 super mini but have not succeeded with grainfather. I get "test failed with error code()" when I run the push test and haven't been able to see where I am wrong as this doesn't give any clues.
the other thing I am struggling with is getting data from the super mini to the gateway, I'm not sure how the transfer is conducted, does it have to have  bluetooth enabled? or is there some other setting up to do with the wifi direct, I have the device name and password on both the super mini device and the gateway but nothing happens.
I have sorted out my charging problem, the charger needs the same family of chips to work so am waiting to see what charger works best when my second option charger arrives, the first I obtained works well but comes with both a charger and receiver which it's wasteful and more expensive.
regards

Dave Ashmore
-- 
Sent from my Android phone with K-9 Mail. Please excuse my brevity.

dtashmore547 commented:
hi Magnus
I don't know what changed but I am now seeing data on the gateway, left it after a couple of hours with nothing happening, came to it today and the screen is full of data. 
I still would like to get the push to grainfather working as it is a very useful app and not continuously nagging for upgrade whilst offering multiple floating hydrometer devices. any pointers on this subject would be gratefully accepted. a year or so back I had an Ispindel connected to grainfather but cannot remember what I did and where it differs from connecting the gravitymon software.
regards
David Ashmore

On 17 March 2025 18:17:12 GMT+02:00, ***@***.*** wrote:
>good day Magnus
>I am finally back on the job after a few weeks and two shoulder operations. 
>I am having some difficulty with pushing to grainfather, have you any experience with this site? I have brewspy and brewfather working on my c3 super mini but have not succeeded with grainfather. I get "test failed with error code()" when I run the push test and haven't been able to see where I am wrong as this doesn't give any clues.
>the other thing I am struggling with is getting data from the super mini to the gateway, I'm not sure how the transfer is conducted, does it have to have  bluetooth enabled? or is there some other setting up to do with the wifi direct, I have the device name and password on both the super mini device and the gateway but nothing happens.
>I have sorted out my charging problem, the charger needs the same family of chips to work so am waiting to see what charger works best when my second option charger arrives, the first I obtained works well but comes with both a charger and receiver which it's wasteful and more expensive.
>regards
>
>Dave Ashmore
>-- 
>Sent from my Android phone with K-9 Mail. Please excuse my brevity.
-- 
Sent from my Android phone with K-9 Mail. Please excuse my brevity.

mp-se commented:
I have not used or tested grainfather but if you had it running with ispindel then it should work with gravitymon as well. It would help to get some output from the serial when you try the push test to see what the errors are. 

The one thing that comes to mind is the format template, gravitymon has a few new parameters that grainfather might not like. You can try to choose the ispindle format template that is an exact match. This should not matter when using JSON unless they dont do proper parsing of the data on their end. Please open a new issue for this and dont comment on a closed PR.

dtashmore547 commented:
hi Magnus
thanks for the info, what the gravitymon device wasn't telling me was the error code but this did show on the serial monitor, it was due to trying to send data too often. having waited for enough time to overcome this I found that I needed the ispindel format and it worked fine.
also I have 4 C3-super-mini devices all communicating via the gravitymon-gateway to my homeassistant server which is brilliant. thanks for the great work.
I will try my best to get the build onto your github repository as soon as I am happy with the charger device. the board, components and build info is on pcbway and I will get it finished and linked soon.
regards
David Ashmore

On 19 March 2025 09:05:30 GMT+02:00, Magnus ***@***.***> wrote:
>mp-se left a comment (mp-se/gravitymon#205)
>
>I have not used or tested grainfather but if you had it running with ispindel then it should work with gravitymon as well. It would help to get some output from the serial when you try the push test to see what the errors are. 
>
>The one thing that comes to mind is the format template, gravitymon has a few new parameters that grainfather might not like. You can try to choose the ispindle format template that is an exact match. This should not matter when using JSON unless they dont do proper parsing of the data on their end. Please open a new issue for this and dont comment on a closed PR. 
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/mp-se/gravitymon/pull/205#issuecomment-2735545610
>You are receiving this because you authored the thread.
>
>Message ID: ***@***.***>
-- 
Sent from my Android phone with K-9 Mail. Please excuse my brevity.

dtashmore547 commented:
hi Magnus
i am still busy evaluating the super mini module for its use in the floating hydrometer, definitely not as straight forward as i initially expected, thanks to your excellent gravitymon gateway device i can now properly evaluate my four built devices and compare them to each other and expectations. the first thing i found was that the regulator onboard the devices is not suitable for low power and low input voltage, tends to drop out around 3.6V and isn't terribly good power wise. fortunately the gyro is especially good in this respect as it is intended for low power and lithium batteries so i have removed the super mini regulator and bridged from the gyro regulator to get the 3.3V supply, this has cured the problem. 
 i had an unexpected problem with my logging the other day, after 4 days problem free the setup suddenly stopped logging to my homeassistant server, the super mini devices were still sending as observed by the onboard LED but the gateway wasn't responding, i needed to reboot the gateway to get it working again, have you any ideas what may have happened? i have the device on a dedicated battery backup so no power issues.
regards
Dave

On 22 January 2025 14:43:59 GMT+02:00, ***@***.*** wrote:
>i will redo thanks, i will also check disabling ble though i would like this as i am awaiting parts to build the Bluetooth bridge. will have to wait until tomorrow as my shoulder is nagging at me now, doesn't take much.
>regards
>Dave
>
>On 22 January 2025 13:07:32 GMT+02:00, Magnus ***@***.***> wrote:
>>Check that you have pulled the dev branch correctly because the automated build complies but i cannot test it since i dont have the board
>>
>>-- 
>>Reply to this email directly or view it on GitHub:
>>https://github.com/mp-se/gravitymon/pull/205#issuecomment-2606943765
>>You are receiving this because you authored the thread.
>>
>>Message ID: ***@***.***>
>-- 
>Sent from my Android phone with K-9 Mail. Please excuse my brevity.
-- 
Sent from my Android phone with K-9 Mail. Please excuse my brevity.

mp-se commented:
Which version of the gateway are you using ? 

When I did some testing of the new 0.7 beta version I noticed that sometimes the network just stopped working for some unknown reason. But I did rewrite the http post handling f the gatway to do the data processing in a background thread instead of the http post handler, that worked much better in my testing. If you are not running the 0.7 beta1 then you could give that a try to see if the stability would improve. Another option could be to reboot the gateway regularly. 

I have not tested the gateway that much but I plan to install one to support my gravitymon devices.

dtashmore547 commented:
hi Magnus
i am using the 0.5 version so i shall upgrade this. 
thanks for the reply. 
Dave

On 20 April 2025 15:04:31 GMT+02:00, Magnus ***@***.***> wrote:
>mp-se left a comment (mp-se/gravitymon#205)
>
>Which version of the gateway are you using ? 
>
>When I did some testing of the new 0.7 beta version I noticed that sometimes the network just stopped working for some unknown reason. But I did rewrite the http post handling f the gatway to do the data processing in a background thread instead of the http post handler, that worked much better in my testing. If you are not running the 0.7 beta1 then you could give that a try to see if the stability would improve. Another option could be to reboot the gateway regularly. 
>
>I have not tested the gateway that much but I plan to install one to support my gravitymon devices. 
>
>
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/mp-se/gravitymon/pull/205#issuecomment-2817161363
>You are receiving this because you authored the thread.
>
>Message ID: ***@***.***>
-- 
Sent from my Android phone with K-9 Mail. Please excuse my brevity.

mp-se commented:
0.7 works with both ble and http

dtashmore547 commented:
hi Magnus
I tried again with 0.7 but again it doesn't work, I get no display and the serial port only sends out ble info and nothing else. I tried from the zip file and cloning. I didn't erase the eeprom this time as it didn't work last time either.
any idea what could be the reason?
regards
Dave

On 21 April 2025 22:36:42 GMT+02:00, Magnus ***@***.***> wrote:
>mp-se left a comment (mp-se/gravitymon#205)
>
>0.7 works with both ble and http
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/mp-se/gravitymon/pull/205#issuecomment-2819453791
>You are receiving this because you authored the thread.
>
>Message ID: ***@***.***>
-- 
Sent from my Android phone with K-9 Mail. Please excuse my brevity.

mp-se commented:
Dont really know, please open an issue in the gateway project and include the hardware option you are using so i can troubleshot it

dtashmore547 commented:
hi Magnus
is there something I should be doing when building the gateway version 0.7, I only see 3 files, main.cpp, ble_gateway.cpp and ble_gateway.hpp. there are no other source filled as there are with v0.5 and 0.6.
also I am having problems with V0.6 as it will not go beyond synchronizing time on the display, and therefore doesn't acknowledge and display the gravitymon devices I am testing.
there seems to be a wifi issue as I am connected to it on my wifi but when I go into the wifi settings it hangs with a whirley in the top right hand corner. It also hangs when trying to do a push test to my homeassistant server.
I am running it on an esp32-wroom device for testing.
regards
Dave

On 23 April 2025 16:06:17 GMT+02:00, ***@***.*** wrote:
>hi Magnus
>I tried again with 0.7 but again it doesn't work, I get no display and the serial port only sends out ble info and nothing else. I tried from the zip file and cloning. I didn't erase the eeprom this time as it didn't work last time either.
>any idea what could be the reason?
>regards
>Dave
>
>On 21 April 2025 22:36:42 GMT+02:00, Magnus ***@***.***> wrote:
>>mp-se left a comment (mp-se/gravitymon#205)
>>
>>0.7 works with both ble and http
>>
>>-- 
>>Reply to this email directly or view it on GitHub:
>>https://github.com/mp-se/gravitymon/pull/205#issuecomment-2819453791
>>You are receiving this because you authored the thread.
>>
>>Message ID: ***@***.***>
>-- 
>Sent from my Android phone with K-9 Mail. Please excuse my brevity.
-- 
Sent from my Android phone with K-9 Mail. Please excuse my brevity.

mp-se commented:
The code is in the gravitymon repo since most of the code is common, i will update the docs on that
